### PR TITLE
Feature/alex/send email

### PR DESCRIPTION
**Pull request by alexdelicato**
# 
## Proposed changes

Branch with feature for queue that successively gets job id from DB (and from this the user's email and the results report)and sends an email to the user with their corresponding report. 

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] DevOps related

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [project guidelines](https://github.com/Visual-Testing-for-Android-Apps/Project-guidelines) doc
- [x] I locally linted my code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by alexdelicato** [comment]
Also note the calls to the DB may need to be changed to match DB set-up. Waiting on confirmation before removing from draft pr.

File: server/sendEmail/src/send-email.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    */
+
+		console.log("sending email...");
+		// send email with defined transport object
+		const info = await transporter.sendMail({
+			from: process.env.EMAIL,
+			to: emailString,
+			subject: "Vision Job Results",
+			text: "",
+			html: htmlResult,
```
Comment (reteppeter) [inline server/sendEmail/src/send-email.ts:78]: This is just the HTML format of the email, it expects an html string (like ```"<p>Hi</p>"```. The link would still show up if put here, but the email would look weird, so it's better to just put it in the text field.

**Review by reteppeter — CHANGES_REQUESTED** [review]

File: server/sendEmail/src/send-email.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+			id: { S: key },
+		},
+	};
+	const res = await awaitJob(request); // wait for job
+
+	// check if result exists with email and link, will add: && res?.link?.S
+	if (res?.email?.S != null) {
+		// get email and link from job
+		const emailString = res.email.S;
+		const htmlResult = res.link.S;
```
[the remark this answers is not in the document]
Comment (alexdelicato) [reply server/sendEmail/src/send-email.ts]: yep, I just needed to update the code with the proper db reference - should be done now, thanks.

File: server/sendEmail/src/send-email.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+		const transporter = nodemailer.createTransport({
+			host: hostname,
+			port: 465, // secure -> 465, not secure -> 587
+			secure: true,
+			requireTLS: true,
+			auth: {
+				user: process.env.EMAIL,
+				pass: process.env.PASSWORD,
+			},
+			logger: true,
```
[the remark this answers is not in the document]
Comment (alexdelicato) [reply server/sendEmail/src/send-email.ts]: good pick up thanks!

File: server/sendEmail/src/send-email.ts (hunk 1 above)
Comment (alexdelicato) [reply server/sendEmail/src/send-email.ts]: I just went through and cleaned this up so the email has a bit of formatting and tells the user what the email actually is, should be in the latest commit.

**Review by reteppeter — APPROVED** [review]

File: server/postProcessHandler/jest.config.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 module.exports = {
-	roots: ["<rootDir>/src"],
```
[the remark this answers is not in the document]
Comment (josh-sharp) [reply server/postProcessHandler/jest.config.js:2]: How do you think it should be formatted? Given that files are formatted this way across the project, is it possible that someone else wants files to be formatted this way? (or is it just the auto-formatter being annoying)

File: README.md [hunk 5]
```diff
@@ -47,25 +56,23 @@ POST https://knfxd86hz7.execute-api.ap-southeast-2.amazonaws.com/Prod/job
...
-
-✅ Manual trigger owlEye and seenomaly 
+✅ Manual trigger owlEye and seenomaly
 In your AWS console,
-1. go to SQS 
-2. select `send and reieve messages`. 
-Example body format
+
+1. go to SQS
+2. select `send and reieve messages`.
```
Comment (josh-sharp) [inline README.md:65]: Obviously doesn't really matter but a small typo here
```suggestion
2. select `send and receive messages`.
```

**Review by josh-sharp — APPROVED** [review]

**Pull request merged**